### PR TITLE
HOSTEDCP-1215: Use the same etcd snapshot for all replicas during etcd restore

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #[HOSTEDCP-1215](https://issues.redhat.com/browse/HOSTEDCP-1215)
Fixes https://github.com/openshift/hypershift/issues/2736

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by sjenning** [comment]
/hold
until look at alternatives to breaking API compat and see if anyone is using this

File: api/v1alpha1/hostedcluster_types.go [hunk 1]
```diff
@@ -1643,15 +1643,15 @@ type ManagedEtcdStorageSpec struct {
...
+	// RestoreSnapshotURL allows an optional URL to be provided where
 	// an etcd snapshot can be downloaded, for example a pre-signed URL
-	// referencing a storage service, one URL per replica.
+	// referencing a storage service.
 	// This snapshot will be restored on initial startup, only when the etcd PV
 	// is empty.
 	//
 	// +optional
 	// +immutable
-	RestoreSnapshotURL []string `json:"restoreSnapshotURL,omitempty"`
```
Comment (enxebre) [inline api/v1alpha1/hostedcluster_types.go:1654]: limit the slice to len 1 seems more sensible to me, so we don't change the API and we keep our ability to change this in future without needed new field or breaking clients

File: control-plane-operator/controllers/hostedcontrolplane/etcd/reconcile.go [hunk 2]
```diff
@@ -166,14 +166,12 @@ func ReconcileStatefulSet(ss *appsv1.StatefulSet, p *EtcdParams) error {
...
-		c.Env = []corev1.EnvVar{}
-		for i := 0; i < p.DeploymentConfig.Replicas; i++ {
-			c.Env = append(c.Env, corev1.EnvVar{
-				Name:  fmt.Sprintf("RESTORE_URL_ETCD_%d", i),
-				Value: p.StorageSpec.RestoreSnapshotURL[i],
-			})
+		c.Env = []corev1.EnvVar{
+			{
+				Name:  "RESTORE_URL_ETCD",
+				Value: p.StorageSpec.RestoreSnapshotURL[0], // RestoreSnapshotURL can only hve 1 entry
```
Comment (jparrill) [inline control-plane-operator/controllers/hostedcontrolplane/etcd/reconcile.go:172]: - What happens if we have multiples ETCD pods (HA)?
- The backup will take only one and upload that one or what will be the behaviour?

**Review by jparrill — COMMENTED** [review]
Dropped a comment about the behaviour change. Could you please add more detail in the commit message or the PR (or both)? Just to have more context.

File: control-plane-operator/controllers/hostedcontrolplane/etcd/reconcile.go (hunk 2 above)
Comment (muraee) [reply control-plane-operator/controllers/hostedcontrolplane/etcd/reconcile.go:172]: there is no behavior change either in HA or normal mode. We just need a single snapshot for all etcd members, and this PR enforces that through the API.

File: docs/content/how-to/aws/etc-backup-restore.md [hunk 3]
```diff
@@ -17,7 +17,7 @@ Now stop all etcd-writer deployments:
 
     oc scale deployment -n ${HOSTED_CLUSTER_NAMESPACE} --replicas=0 kube-apiserver openshift-apiserver openshift-oauth-apiserver
 
-You can now perform an etcd snapshot via `exec` in each etcd container:
+You can now perform an etcd snapshot via `exec` in any etcd container:
```
Comment (muraee) [inline docs/content/how-to/aws/etc-backup-restore.md:20]: @jparrill you can create the snapshot from **_any_** etcd pod.

**Pull request merged**